File: Compiler/testdata/native/printer.java
class StringUtils {
	public static native String create();
	public static native void printStatic(String test);

	public static void main(String[] args) {
		String string = StringUtils.create();
		StringUtils.printStatic(string);
		string.print();
	}
}

class String {
	public native void print();
}

File: Compiler/testdata/native/printer.c
#include <stdlib.h>
#include <stdio.h>

char*  __attribute__((sysv_abi))  StringUtils$create() {
	char *test = calloc(12, 1);
	sprintf(test, "Hallo\n");
	return test;
}

void  __attribute__((sysv_abi))  StringUtils$printStatic(char *pointer) {
	printf("static: %s", pointer);
}

void  __attribute__((sysv_abi))  String$print(char* this) {
	printf("nonStatic: %s", this);
}
